hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/RecordStore.java

```diff
@@ -314,5 +310,18 @@
      **/
     void maybeDoInitialLoad();
 
+    Storage createStorage(RecordFactory<R> recordFactory, InMemoryFormat memoryFormat);
```

Comment:
There is no reason this method should be in the interface. The AbstractRecordStore wants to create the Storage, and you want to define this method in the sub class. 

1: Just make this method a protected abstract method which only needs to be seen by the AbstractRecordStore subclasses. There is no need to expose this in the API since nobody else apart from the AbstractRecordStore subclass wants to call this method.

2: Such methods can be very dangerous because due to the callback in the constructor, it can happen that the implementing methods sees a partially constructed object. I often prefer to inject the right Storage through the constructor; thereby following DI.